### PR TITLE
Be 52 data service

### PR DESCRIPTION
**Pull request by zafertaskent**
Data service is ready. DataDetail object is added. 

The following is a curl comment to insert data. 

@ogzhnozlci @berkantesk please review.

`curl --insecure -X POST -H 'Content-Type: application/json' \
                                                                                         http://localhost:1337/api/data \
                                                                                         -d '{
                                                                                           "department":"570f4ca7d4e33fee189f973f",
                                                                                           "indicator":"570f516dd4e33fee189f9760",
                                                                                           "year":"2010",
                                                                                           "value":10,
                                                                                           "answers":[
                                                                                             {
                                                                                                 "indicator":"570f516dd4e33fee189f9760",
                                                                                                 "criteria":"570f519bd4e33fee189f9762",
                                                                                                 "question":"570f51c1d4e33fee189f9766"
                                                                                             }
                                                                                           ]
                                                                                             }'`

### CODE REVIEW

File: api/controllers/DataController.js [hunk 1]
```diff
@@ -10,24 +10,37 @@ module.exports = {
...
+  find:function(req,res){
+    console.log("Fİnd all");
+    Data.find().populate("department").populate("indicator").populate("answers").exec(function(err,data){
+      res.json(data);
+    });
+  },
 
+  findOne:function(req,res)
+  {
+    console.log("get one");
```
Comment (adilatilgan) [inline api/controllers/DataController.js]: Please remove unnecessary logs

File: api/controllers/DataController.js [hunk 2]
```diff
@@ -10,24 +10,37 @@ module.exports = {
   chart: function (req, res) {
     Data.find({
       indicator: req.param('indicator'),
-    }).sort('year ASC').populate('department').populate('indicator').exec(function (err, datas) {
+    }).sort('year ASC').populate('department').populate('indicator').populate('answers').exec(function (err, datas) {
       console.log(datas);
```
Comment (adilatilgan) [inline api/controllers/DataController.js]: Please remove unnecessary logs

File: api/controllers/DataDetailController.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+module.exports = {
+	findAll: function (req, res) {
+		console.log("Layn1");
+		DataDetail.find().populate('criteria').populate('indicator').populate('question').exec(function (err, datadetails) {
+			res.json(datadetails);
+		});
+	},
+	find: function (req, res) {
+		console.log("Layn2");
```
Comment (adilatilgan) [inline api/controllers/DataDetailController.js]: Please remove unnecessary logs

File: api/policies/authenticated.js [hunk 4]
```diff
@@ -11,6 +11,7 @@ module.exports = function(req, res, next) {
   // Todo : Read Them from cache
   Authorization.findOne(req.path + ':' + req.method).exec(function(err, auth) {
     //if End point found check authorization
+    return next();
```
Comment (adilatilgan) [inline api/policies/authenticated.js]: Why did you bypass authorization

File: config/routes.js [hunk 5]
```diff
@@ -50,7 +49,7 @@ module.exports.routes = {
     controller: 'AuthController',
     action: 'callback'
   },
-  'get /api/data/:indicator': {
+  'get /api/data/indicator/:indicator': {
```
Comment (adilatilgan) [inline config/routes.js]: @berkantesk @ogzhnozlci  must know about this. Endpoint URL changed to 
 /api/data/indicator/:indicator

**Comment by zafertaskent** [comment]
Fixed all

**Comment by adilatilgan** [comment]
LGTM now, merging

**Pull request merged**